code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1610,87 +1610,240 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 //////////////////////////
 
 /**
- * environment handler for species
+ * Environment handler for species
  *
  * vars:
- * * environment The environment gas mix
- * * H The mob we will stabilize
+ * * environment (required) The environment gas mix
+ * * humi (required)(type: /mob/living/carbon/human) The mob we will target
  */
-/datum/species/proc/handle_environment(datum/gas_mixture/environment, mob/living/carbon/human/H)
-	var/areatemp = H.get_temperature(environment)
+/datum/species/proc/handle_environment(datum/gas_mixture/environment, mob/living/carbon/human/humi)
+	handle_environment_pressure(environment, humi)
 
-	if(H.stat != DEAD) // If you are dead your body does not stabilize naturally
-		natural_bodytemperature_stabilization(environment, H)
+/**
+ * Body temperature handler for species
+ * vars:
+ * * humi (required)(type: /mob/living/carbon/human) The mob we will target
+ */
+/datum/species/proc/handle_body_temperature(mob/living/carbon/human/humi)
+	if(humi.stat == DEAD) // If you are dead none of this matters
+		return
+
+	body_temperature_core(humi)
+	body_temperature_skin(humi)
+	body_temperature_alerts(humi)
+	body_temperature_damage(humi)
+
+/**
+ * Used to stabilize the core temperature back to normal on living mobs
+ *
+ * The metabolisim heats up the core of the mob trying to keep it at the normal body temp
+ * vars:
+ * * humi (required) The mob we will stabilize
+ */
+/datum/species/proc/body_temperature_core(mob/living/carbon/human/humi)
+	var/natural_change = get_temp_change_amount(humi.get_body_temp_normal() - humi.coretemperature, 0.12)
+	humi.adjust_coretemperature(humi.metabolism_efficiency * natural_change)
+
+/**
+ * Used to normalize the skin temperature on living mobs
+ * vars:
+ * * humi (required) The mob we will targeting
+ */
+/datum/species/proc/body_temperature_skin(mob/living/carbon/human/humi)
+
+	// change the core based on the skin temp
+	var/skin_core_diff = humi.bodytemperature - humi.coretemperature
+	var/skin_core_change = get_temp_change_amount(skin_core_diff, 0.08)
+
+	humi.adjust_coretemperature(skin_core_change)
+
+	// get the enviroment details
+	var/datum/gas_mixture/environment = humi.loc.return_air()
```

Comment:
Shouldn't you be getting this from a turf? I suppose it would effect stuff like hiding in lockers during fires, etc.

Reply:
This is how handle_environment gets the local air earlier in the life proc, getting this here prevents us from passing the value through several procs that did not need it directly.